server/service/src/sync/translations/invoice_line.rs

```diff
@@ -219,12 +219,11 @@ impl SyncTranslation for InvoiceLineTranslation {
         // TODO: remove the stock_line_is_valid check once central server does not generate the inbound shipment
         // omSupply should be generating the inbound, with valid stock lines.
         // Currently a uuid is assigned by central for the stock_line id which causes a foreign key constraint violation
-        let is_stock_line_valid = match stock_line_id {
-            Some(ref stock_line_id) => StockLineRowRepository::new(connection)
-                .find_one_by_id(stock_line_id)?
-                .is_some(),
-            None => true,
-        };
+        let is_stock_line_valid = stock_line_id.as_ref().is_some_and(|stock_line_id| {
```

Comment:
Will the `is_some_and` fn call return false if `None`, not integrating records if there is no ID?

The first workflow that I think would break for this is that if you are reinitialising a site that that has some amount of "New" inbound shipments with lines. No stock affected yet, so no stock_lines relating to the invoice_lines.

Reply:
Yes, that's what I understood from the docs. There's code below that saves None for stock line if not valid... do you think it will still break?

```js
       let stock_line_id = if is_record_active_on_site && is_stock_line_valid {
            stock_line_id
        } else {
            None
        };
```

Reply:
nah yea that's fine 👍